### PR TITLE
ruby26.y: Endless ranges support.

### PR DESCRIPTION
**Pull request by iliabylich**
This commit tracks upstream commit ruby/ruby@7f95eed.
Closes https://github.com/whitequark/parser/issues/504

### CODE REVIEW

**Comment by whitequark** [comment]
Can you update `AST_FORMAT.md` too?

**Comment by iliabylich** [comment]
@whitequark Done

File: doc/AST_FORMAT.md [hunk 1]
```diff
@@ -322,6 +322,20 @@ Format:
...
+
+Can be inclusive or exclusive.
+
+Format:
+
+~~~
+(irange (int 1) nil)
+"1.."
+  ~~ operator
+ ~~~ expression
```
Comment (whitequark) [inline doc/AST_FORMAT.md]: I try to list all forms explicitly in this document, i.e. both `irange` and `erange`.

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**